src/components/pages/landing/Gooddollar.js

```diff
@@ -29,96 +30,104 @@ export const Gooddollar = ({ setShowGooddollarVerification }) => {
     r: ['name'],
     rdu: window.location.href,
   });
+  const [gooddollarData, setGooddollarData] = React.useState(null);
   const [rawGoodDollarData, setRawGoodDollarData] = React.useState(null);
   const [editableFields, setEditableFields] = React.useState({
     name: true,
     gDollarAccount: true,
     status: true,
   });
-  const [isVerifyApiError, setIsVerifyApiError] = React.useState(false);
   const [submitting, setSubmitting] = React.useState(null);
 
-  const { values, handleSubmit, handleBlur, handleChange, setValues } =
-    useFormik({
-      initialValues: {
-        name: '',
-        gDollarAccount: '',
-        status: '',
-      },
-      validationSchema: Yup.object().shape({
-        email: Yup.string().email('Invalid email'),
-      }),
-      onSubmit: async (data) => {
-        // here need to clear url and remove all unnecessary data from url for near wallet redirect
-        window.history.replaceState({}, '', window.location.origin);
-        setSubmitting(true);
-        const { sig, ...rawData } = rawGoodDollarData;
+  const {
+    values,
+    handleSubmit,
+    handleBlur,
+    handleChange,
+    setValues,
+    errors,
+    setFieldValue,
+  } = useFormik({
+    initialValues: {
+      name: '',
+      gDollarAccount: '',
+      status: '',
+    },
+    validationSchema: Yup.object().shape({
+      email: Yup.string().email('Invalid email'),
+    }),
+    onSubmit: async (data) => {
```

Comment:
this is a long function and should be broken down.

Reply:
Will do in a separate pr